sanitize.go

```diff
@@ -0,0 +1,143 @@
+package dns
+
+import "strings"
+
+// Dedup removes identical RRs from rrs. It preserves the original ordering.
+// The lowest TTL of any duplicates is used in the remaining one.
+//
+// TODO(miek): This function will be extended to also look for CNAMEs and DNAMEs.
```

Comment:
I wonder if this should be optional... or in another function now... Not sure where this would be used.